### DISCUSSION TITLE
主应用访问微应用的时候,微应用服务掉线,主应用会报错,请问我该怎么捕获这个错误?并处理他

### DISCUSSION THREAD

**Discussion by Lands-1203** [umijs/umi#10824] [Q&A]
![image](https://user-images.githubusercontent.com/61341868/227115656-224abd8f-a02b-411d-acc5-e8cea13d970f.png)
我把微应用服务停了
从主应用访问会直接报错,访问丢失的时候我想用一个页面承接

**Comment by fz6m** [comment]
没有资源肯定会报错的，在生产里不会有这个 error overlay 。

开发时这个 overlay 你可以手动点右上角叉把它关掉，切回正常的路由。

自定义加载失败展示内容参考这里：https://umijs.org/docs/max/micro-frontend#%E8%87%AA%E5%AE%9A%E4%B9%89%E9%94%99%E8%AF%AF%E6%8D%95%E8%8E%B7%E7%BB%84%E4%BB%B6

**Reply by Lands-1203** [reply]
> 没有资源肯定会报错的，在生产里不会有这个 error overlay 。
> 
> 开发时这个 overlay 你可以手动点右上角叉把它关掉，切回正常的路由。
> 
> 自定义加载失败展示内容参考这里：https://umijs.org/docs/max/micro-frontend#%E8%87%AA%E5%AE%9A%E4%B9%89%E9%94%99%E8%AF%AF%E6%8D%95%E8%8E%B7%E7%BB%84%E4%BB%B6

这个错误会出现在生产环境吗?

**Reply by fz6m** [reply]
生产里会默认展示子应用错误栈的消息，但网络失败路由里啥也不会有，看上面的链接自定义自己的错误承接展示内容。

你可以在 devtools 里 block 掉子应用的 html 加载本地模拟下加载失败会怎么样

**Reply by Lands-1203** [reply]
> 

似乎无法捕获
![image](https://user-images.githubusercontent.com/61341868/227413137-23d8fec8-2961-4842-8873-71ddeed68b5c.png)
![image](https://user-images.githubusercontent.com/61341868/227413419-b0668daa-21c7-460b-a801-324645ab2778.png)
![image](https://user-images.githubusercontent.com/61341868/227413493-e00e03bc-2c54-41f2-ad31-5dddf0986ddb.png)



并且还有一个问题 （这个问题不是重点，上面的捕获才是）在config中从page导出组件无法启动

**Reply by Lands-1203** [reply]
> 生产里会默认展示子应用错误栈的消息，但网络失败路由里啥也不会有，看上面的链接自定义自己的错误承接展示内容。
> 
> 你可以在 devtools 里 block 掉子应用的 html 加载本地模拟下加载失败会怎么样


主应用：https://github.com/Lands-1203/master-application-template-umi.git
微应用：https://github.com/Lands-1203/micro-application-template-umi.git

**Reply by fz6m** [reply]
新建一个路由页面，用 `<MicroApp />` 渲染配置，routes.ts 里不支持 tsx 配置。

```tsx
import CustomErrorBoundary from '@/components/CustomErrorBoundary';
import { MicroApp } from 'umi';
 
export default () => {
  return (
    <MicroApp
      name="app1"
      errorBoundary={(error) => <CustomErrorBoundary error={error} />}
    />
  );
};
```

**Reply by Lands-1203** [reply]
> 新建一个路由页面，用 `<MicroApp />` 渲染配置，routes.ts 里不支持 tsx 配置。
> 
> ```tsx
> import CustomErrorBoundary from '@/components/CustomErrorBoundary';
> import { MicroApp } from 'umi';
>  
> export default () => {
>   return (
>     <MicroApp
>       name="app1"
>       errorBoundary={(error) => <CustomErrorBoundary error={error} />}
>     />
>   );
> };
> ```

使用<MicroApp /> 组件引入子应用可以实现。
那文档中是不是写错了？
![image](https://user-images.githubusercontent.com/61341868/227424711-685cf7c2-ab3a-42f8-b2be-3e0b366ff75c.png)
